python/ray/data/impl/stats.py

```diff
@@ -351,3 +351,16 @@ def summary_string(self, exclude_first_window: bool = True) -> str:
         out += "* Time in user code: {}\n".format(fmt(self.iter_user_s.get()))
         out += "* Total time: {}\n".format(fmt(self.iter_total_s.get()))
         return out
+
+
+class _StatsActorWrapper:
```

Comment:
Hmm another option is we move StatsActor into the global DatasetContext. We already have the DesignatedBlockOwner actor, maybe they can be merged?

Reply:
I looked at doing that, but thought that it might blow up the scope of the PR without much benefit. Looking again, it seems doable if we remove the stats actor handle from `DatasetStats` and from the read tasks, and always pull it from the current context but we'll have to explicitly pass the context to the read task to prevent the existing issue with capturing actor handles in closures.

Note that we'll still need a custom serialization hook for `DatasetContext`, which will be a bit more complex than serializing this little actor handle wrapper. I'll try making this change and see if it's worth it.

Reply:
Saving this for a follow-up PR along with the moving of the read remote function to module scope as done in [this PR](https://github.com/ray-project/ray/pull/22580).